internal/core/software_renderer.rs

```diff
@@ -328,6 +328,33 @@ impl SoftwareRenderer {
     /// Returns the physical dirty region for this frame, excluding the extra_draw_region,
     /// in the window frame of reference. It affected by the screen rotation.
     pub fn render(&self, buffer: &mut [impl TargetPixel], pixel_stride: usize) -> PhysicalRegion {
+        self.render_with_post_render_callback(buffer, pixel_stride, None)
+    }
+
+    /// Render the window to the given frame buffer.
+    ///
+    /// The renderer uses a cache internally and will only render the part of the window
+    /// which are dirty. The `extra_draw_region` is an extra regin which will also
+    /// be rendered. (eg: the previous dirty region in case of double buffering)
+    /// This function returns the region that was rendered.
+    ///
+    /// The pixel_stride is the size, in pixel, between two line in the buffer
+    /// The buffer needs to be big enough to contain the window, so its size must be at least
+    /// `pixel_stride * height`, or `pixel_stride * width` if the screen is rotated by 90°.
+    ///
+    /// Returns the physical dirty region for this frame, excluding the extra_draw_region,
+    /// in the window frame of reference. It affected by the screen rotation.
+    ///
+    /// After the rendering of the scene is complete, the optional post_render_callback
+    /// will be invoked, which provides the caller with access to the ItemRenderer, to render
+    /// additional content.
+    #[doc(hidden)]
+    pub fn render_with_post_render_callback(
+        &self,
+        buffer: &mut [impl TargetPixel],
+        pixel_stride: usize,
+        post_render_callback: Option<&dyn Fn(&mut dyn ItemRenderer)>,
```

Comment:
Not a fan of having that in the pubic API (even if hidden) of the software renderer.
Also because we will never end up "stabilizing" this since it uses the ItemRenderer which is even more private.

I'd prefer if you would just blit the pixel manually in linuxkms on top of the surface.

The alternative would be to move the cursor handling in i-slint-core and having it as an `Image{}` item on top of the scene. I think this may be usefull for other backend too such as the uefi.

Otherwise if you don't have time for that, please add an InternalToken parameter to be sure.

Reply:
Agreed. I'll try the approach with an Image on top, wrapped in a hand-written ItemTree and rendered inline like an inline popup on top of the scene, using the slice passed to draw_contents.